src/cloud_provider/aws/router.rs

```diff
@@ -282,34 +288,98 @@ impl Service for Router {
 
 impl crate::cloud_provider::service::Router for Router {
     fn check_domains(&self) -> Result<(), EngineError> {
-        let check_result = retry::retry(Fibonacci::from_millis(3000).take(1), || {
-            // TODO send information back to the core
-            info!("check custom domain {}", self.default_domain.as_str());
-            match lookup_host(self.default_domain.as_str()) {
-                Ok(_) => OperationResult::Ok(()),
+        let listeners_helper = ListenersHelper::new(&self.listeners);
+
+        listeners_helper.start_in_progress(ProgressInfo::new(
+            ProgressScope::Router {
+                id: self.id().into(),
+            },
+            ProgressLevel::Info,
+            Some(format!(
+                "Let's check that {} domain is ready. Please wait, it can take some time...",
+                self.name_with_id()
+            )),
+            self.context.execution_id(),
+        ));
+
+        let mut resolver_options = ResolverOpts::default();
+        resolver_options.cache_size = 0;
+        resolver_options.use_hosts_file = false;
+
+        let resolver = match Resolver::new(ResolverConfig::google(), resolver_options) {
```

Comment:
Please use  cloudflare() instead. It's more relevant due to our config

Reply:
Checking through google is maybe a better idea due to the fact most users/isp may use Google resolver instead of Cloudflare. Don't you think?

Reply:
You're right, but it may take more time before being available. Let's see if we encounter timeouts. If it's the case, we'll leverage the retry time or switch to cloudflare.